addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -2260,6 +2260,64 @@ QUnit.module('relational_fields', {
 
     QUnit.module('FieldOne2Many');
 
+    QUnit.test('one2many list editable with cell readonly modifier', function (assert) {
+        assert.expect(4);
+
+        this.data.partner.records[0].p = [2];
+        this.data.partner.records[1].turtles = [1,2];
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch:'<form string="Partners">' +
+                    '<field name="p">' +
+                        '<tree editable="bottom">' +
+                            '<field name="turtles" invisible="1"/>' +
+                            '<field name="foo" attrs="{&quot;readonly&quot; : [(&quot;turtles&quot;, &quot;!=&quot;, [])] }"/>' +
+                            '<field name="qux" attrs="{&quot;readonly&quot; : [(&quot;turtles&quot;, &quot;!=&quot;, [])] }"/>' +
+                        '</tree>' +
+                    '</field>' +
+                '</form>',
+            res_id: 1,
+            mockRPC: function (route, args) {
+                if (route === '/web/dataset/call_kw/partner/write') {
+                    assert.deepEqual(args.args[1].p[1][2], {foo: 'ff', qux: 999},
+                        'The right values should be written');
+                }
+                return this._super(route, args);
+            }
+        });
+
+        form.$buttons.find('.btn.btn-primary.btn-sm.o_form_button_edit').click();
+        form.$('.o_field_x2many_list_row_add a').click();
+
+        var $target_input = $('.o_data_row.o_selected_row .o_field_char.o_field_widget.o_input[name=foo]');
+        assert.equal($target_input[0], document.activeElement,
+            'The first input of the line should have the focus');
+
+        $target_input.val('f');
+        $target_input.trigger('input');
+        $target_input.val($target_input.val() + 'f');
+        $target_input.trigger('input');
+
+        assert.equal($target_input[0], document.activeElement,
+            'The first input of the line should still have the focus');
+
+        $target_input.trigger($.Event('keydown', {which: 9, keyCode:9}));
+
+        var $second_target = $('.o_data_row.o_selected_row .o_field_float.o_field_number.o_field_widget.o_input[name=qux]');
```

Comment:
here again, a very specific selector is kind of fragile, because any change in the future may break this test. I suggest .o_input[name=qux]

Reply:
also, $secondTarget